octobot_evaluators/evaluators/abstract_evaluator.py

```diff
@@ -86,6 +89,58 @@ def __init__(self, tentacles_setup_config: tm_configuration.TentaclesSetupConfig
         # Define evaluators default consumer priority level
         self.priority_level: int = channel_enums.ChannelConsumerPriorityLevels.MEDIUM.value
 
+        self.consumers = []
+
+        # True when this evaluator is only triggered on closed candles
+        self.is_triggered_after_candle_close = False
+
+        # Other evaluators that might have been called by this evaluator.
+        # This evaluator is now responsible for managing their cache
+        self.called_nested_evaluators = set()
+        if post_init:
+            self.post_init(tentacles_setup_config)
+
+    def post_init(self, tentacles_setup_config):
+        """
+        Automatically called after __init__ when post_init is True (default)
+        Override when necessary
+        :param tentacles_setup_config: the tentacles_setup_config __init__ argument
+        :return: None
+        """
+        pass
+
+    @classmethod
+    def factory_with_local_config(
+            cls,
+            tentacles_setup_config: tm_configuration.TentaclesSetupConfiguration,
+            specific_config: dict,
+            post_init=False):
+        evaluator_instance = cls(tentacles_setup_config, post_init=post_init)
+        evaluator_instance.logger = commons_logging.get_logger(evaluator_instance.get_name())
+        evaluator_instance.specific_config = specific_config
+        return evaluator_instance
```

Comment:
Why do we need this to be a class method? It doesn't make sense to me. I think it should be placed outside of AbstractEvaluator

Reply:
When using nested calls inside a script, we need to create a local instance of the required evaluator. This function allows to create it (without reading config files etc).
Right, I will move it into its own file